Setup rotating log file

olivielpeau opened the pull request:
### What does this PR do?

* start logging to an actual log file, in both the agent and dogstatsd. Its rotation is handled by seelog
* add a log level option
* format log entries with a format that's close to the agent 5

### Motivation

During testing, I realized that by default `upstart` rotates the log files only once it's quite huge (> 1GB), which is not reasonable for the agent. With these changes, the agent uses the same rotation settings as the agent5.

### Additional Notes

See also commit descriptions

Logging in the agent 6 in general would deserve much more work and options (log to syslog, etc), which is out of the scope of this PR.

olivielpeau committed:
[agent][logging] Use a rotating file for logging

For the agent binary. The log file path can be configured, and has
a sensible default.

olivielpeau committed:
[logging] Remove upstart stdout redirection to file

Now that the agent can log directly to a file

olivielpeau committed:
[dogstastsd][logging] Bring file logging to dogstatsd

olivielpeau committed:
[config/log] Add comment on SetupLogger

truthbk reviewed (approved):
This looks good to me. I added a comment in case you also want to implement the log level configuration in this PR, I think it would make sense. We can definitely do it in a different PR if you prefer.

olivielpeau committed:
[log] Add basic support of log levels

Support for log levels in uppercase to accomodate agent 5 log levels

olivielpeau committed:
[log] Document log options

And change log_file parameter name to a common name (a dogstatsd binary
wouldn't be using the same config file as an agent binary anyway)

olivielpeau committed:
[log] Add readable format

Close to agent 5 log format

truthbk reviewed (approved):
Looks great! 👍
